pkg/vault/contracts/test/RouterCommonMock.sol

```diff
@@ -41,14 +41,6 @@ contract RouterCommonMock is RouterCommon {
         return _vault;
     }
 
-    function getWeth() external view returns (IWETH) {
-        return _weth;
-    }
-
-    function getPermit2() external view returns (IPermit2 permit2) {
-        return _permit2;
-    }
-
```

Comment:
These are used only in tests, and removing them from here increases gas cost a bit and increases all router sizes. Does it need to be in router common?

Reply:
It's a bit annoying, but I realized that initializing a pool in the same tx it is created is cumbersome without this if you want to use the standard router. The problem is that you need to deal with token transfers, and the factory needs to approve `permit2` besides the router. Then, you need to have `permit2` address available in the pool factory.

It's not ideal, but it might become useful at some point in the future. Initializing in the same tx as creation is not the most common use case, but if by any reason the creator doesn't want the possibility of initialization frontrun, it should be doable.